staging/src/k8s.io/apiserver/pkg/endpoints/filters/trace.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"net/http"
+	"time"
+
+	utiltrace "k8s.io/utils/trace"
+
+	"k8s.io/apiserver/pkg/endpoints/internal"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// WithTrace decorates a http.Handler with tracing for all the non-long running
+// requests coming to the server.
+func WithTrace(handler http.Handler, longRunningCheck request.LongRunningRequestCheck) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		isLongRunning := false
+		if longRunningCheck != nil {
+			ri, ok := request.RequestInfoFrom(req.Context())
+			if ok && longRunningCheck(req, ri) {
+				isLongRunning = true
+			}
+		}
+
+		if !isLongRunning {
+			trace := utiltrace.New("HTTP "+req.Method,
+				utiltrace.Field{Key: "url", Value: req.URL.Path},
+				utiltrace.Field{Key: "user-agent", Value: &internal.LazyTruncatedUserAgent{req}},
+				utiltrace.Field{Key: "client", Value: &internal.LazyClientIP{req}})
+			defer trace.LogIfLong(750 * time.Millisecond)
```

Comment:
is this really the number we used before? Thinking especially about LIST requests which will never finish in that time. Didn't we define the time bound in the endpoint implementations before?

Reply:
we could also have a much larger number here and let the endpoints shrinken it down for all the steps inside the endpoint implementation.

Reply:
Good question.

Did a grep for "LogIfLong" in endpoint and they're all 500ms. create, get, update, patch, delete, and [list](https://github.com/jpbetz/kubernetes/blob/4439b767f5b27a7931a493566233c6672203c2c5/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L277).

I agree 500ms seems low for list.

How about we pick a number that we feel is appropriate for list's and set the filter limit to be maybe 500ms over that number?

Maybe 1s for list and 1.5s for the trace filter?

Reply:
Don't we still have the lower numbers in the endpoints? This is just another bound. If a nested trace exceeds the nested bound, we still get logging, don't we? So even if we put a 30s here, we are fine.

Reply:
Yeah, we do. I'm okay with starting with 30s, I guess. That way we don't risk accidentally flooding the logs with this change.